src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -213,6 +213,25 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::DefaultPr
     return *_userDefaultProfileSettings;
 }
 
+// Method Description:
+// - Create a new profile based off the default profile settings.
+// Arguments:
+// - <none>
+// Return Value:
+// - a reference to the new profile
+winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNewProfile()
+{
+    auto newProfile{ _userDefaultProfileSettings->CreateChild() };
+    _allProfiles.Append(*newProfile);
+
+    // Give the new profile a distinct name so a guid is properly generated
+    auto newName = fmt::format(L"Profile {}", _allProfiles.Size());

```

Comment:
clever

Reply:
You may want to +1 that size because Size is 0-ba... wait a second you already did